datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -780,24 +780,15 @@ impl ExecutionPlan for HashJoinExec {
         // TODO stats: it is not possible in general to know the output size of joins
         // There are some special cases though, for example:
         // - `A LEFT JOIN B ON A.col=B.col` with `COUNT_DISTINCT(B.col)=COUNT(B.col)`
-        let mut stats = estimate_join_statistics(
+        let stats = estimate_join_statistics(
             Arc::clone(&self.left),
             Arc::clone(&self.right),
             self.on.clone(),
             &self.join_type,
             &self.join_schema,
         )?;
         // Project statistics if there is a projection
-        if let Some(projection) = &self.projection {
-            stats.column_statistics = stats
```

Comment:
this code also appears to assume projections never contain repeated values 🤔  I can try and improve the statistics calculation in a future PR to avoid cloning

Reply:
I made a PR that projects the column statistics without copying as well as handling repetitions: https://github.com/apache/datafusion/pull/13225